Composer/packages/server/src/models/bot/botProject.ts

```diff
@@ -199,6 +199,15 @@ export class BotProject implements IBotProject {
   public getEnvSettings = async (obfuscate: boolean) => {
     const settings = await this.settingManager.get(obfuscate);
 
+    // Resolve relative path for custom runtime if the path is relative
+    if (settings?.runtime?.customRuntime && settings.runtime.path && !Path.isAbsolute(settings.runtime.path)) {
```

Comment:
first condition is worried that the settings is undefined, second and third conditions are not?